spec/requests/api/v1/subscriptions_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+RSpec.describe "Api::V1::Subscriptions", type: :request do
+  describe "Subscription API" do
+    let!(:customer) {create(:customer)}
+    let!(:tea) {create(:tea)}
+    
+    
+    describe "POST /api/v1/subscriptions" do
+      it "creates a new subscription" do
+        attrs = {
+          title: "Earl Grey",
+          price: 5.00,
+          status: "active",
+          frequency: "monthly",
+          customer_id: customer.id,
+          tea_id: tea.id
+        }
+
+        post "/api/v1/subscriptions", params: { subscription: attrs }
+        sub_data = JSON.parse(response.body, symbolize_names: true)
+
+        expect(response).to be_successful
+        expect(response.status).to eq(200)
+
+        expect(sub_data).to be_a(Hash)
+        expect(sub_data).to have_key(:data)
+
+        expect(sub_data[:data]).to be_a(Hash)
+        
+        expect(sub_data[:data]).to have_key(:id)
+        expect(sub_data[:data][:id]).to be_a(String)
+  
+        expect(sub_data[:data]).to have_key(:type)
+        expect(sub_data[:data][:type]).to be_a(String)
+
+        expect(sub_data[:data]).to have_key(:attributes)
+        expect(sub_data[:data][:attributes]).to be_a(Hash)
+
+        expect(sub_data[:data][:attributes]).to have_key(:id)
+        expect(sub_data[:data][:attributes][:id]).to be_an(Integer)
+
+        expect(sub_data[:data][:attributes]).to have_key(:title)
+        expect(sub_data[:data][:attributes][:title]).to be_a(String)
+
+        expect(sub_data[:data][:attributes]).to have_key(:price)
+        expect(sub_data[:data][:attributes][:price]).to be_a(Float)
+
+        expect(sub_data[:data][:attributes]).to have_key(:status)
+        expect(sub_data[:data][:attributes][:status]).to be_a(String)
+
+        expect(sub_data[:data][:attributes]).to have_key(:frequency)
+        expect(sub_data[:data][:attributes][:frequency]).to be_a(String)
+
+        expect(sub_data[:data][:attributes]).to have_key(:customer_id)
+        expect(sub_data[:data][:attributes][:customer_id]).to be_an(Integer)
+
+        expect(sub_data[:data][:attributes]).to have_key(:tea_id)
+        expect(sub_data[:data][:attributes][:tea_id]).to be_an(Integer)
+      end
+    end
+    
+    describe "sad path" do
+      it 'returns 422 with invalid attributes if customer and/or tea does not exist' do
+        invalid_attrs = {
```

Comment:
sad path works if I don't include the attributes but if I try to put in an customer id that doesn't exist it will not pass the test locally. when I put a non existant customer id in postman raw body it gives me the expected response though.